Cookbook/Proposals/AdditionalRulesForDanger.md

```diff
@@ -0,0 +1,133 @@
+
+The rules I am about to introduce are already known to us - we apply them on our PR reviews on a daily basis, after all - but, nevertheless, why rely on memory alone when we can automate them?
+
+## Proposed solution
+
+We will now go through the list of suggested rules and consider:
+    a) if they should be applied to our codebase (at all)
+    b) which tool (therefore, which approach) should we use to minimize further occurrences of the problem at hand
+
+### Custom ViewModels in the test suite
+
+Custom ViewModels should not be defined as part of our tests; we should rely on `StubViewModel`s (from `BabylonSnapshotTestUtilities`) instead (although, in theory, setting up custom ViewModels could be useful in particular circumstances).
+
+This rule would be rather simple to enforce via Danger, using the following regex: `class .*ViewModel\W` inside every test file. 
+To do so, we should whitelist every file that matched `Tests.*`.
+
+Nevertheless, as of June 7th, 2019, we have 254 occurences of this rule in 251 files. 
+
+As such, I believe this rule should only be enforced via Danger because using SwiftLint to monitor this would trigger 254 new warnings  (obviously). Since we already have a tremendous amount of warnings (~500), we'd have to refactor all of these new warnigs just to get back to our starting point. This would require a rather significant effort for very little benefit.
+
+On the other hand, enforcing the rule with Danger wouldn't cause those issues but we would only be able to detect new occurences of this anti-pattern since Danger will only inspect changes made in pull requests.
+
+### Monitoring ViewController lifecycle directly instead of relying on `ScreenLifecycleEvent`
+
+With the introduction of `ScreenLifecycleEvent`, developers should no longer manually monitor the viewController's lifecycle events.
+Instead, this kind of monitoring can now easily be done via the `send(_: ScreenLifecycleEvent)` method.
+
+Here's a brief example of how it was typically done in both Builders and FlowControllers:
+
+```swift
+viewController
+            .reactive
+            .signal(for: #selector(UIViewController.viewWillDisappear))
+            .discardValues()
+            .observeValues(dismiss)
+```
+
+Nowadays, this can be done much more elegantly (thanks Anders!):
+
+```swift
+func send(_ event: ScreenLifecycleEvent) {
+        switch (event, state.status) {
+        case (.didAppear, .showing):
+            runSomething()
+        default:
+            break
+        }
+    }
+```
+
+As of June 7th, 2019, we still have 16 results in 13 different files.
+
+In order to detect this, we could simply monitor occurences of the following regex:
+
+`\.signal\(for: \#selector\(UIViewController\.`
```

Comment:
This only addresses part the problem, we have a reasonable number of custom view controllers doing a similar logic too.